library/core/src/iter/traits/iterator.rs

```diff
@@ -3475,8 +3475,8 @@ pub trait Iterator {
         }
     }
 
-    /// [Lexicographically](Ord#lexicographical-comparison) compares the elements of this [`Iterator`] with those
-    /// of another.
+    /// [Lexicographically](Ord#lexicographical-comparison) compares the elements of this [`Iterator`], which
+    /// implements the `PartialOrd` trait, with those of another.
```

Comment:
Hmm, I like the idea that improving the comment here, but I think this isn't quite the right phrasing, since it could be interpreted as the *iterator* implementing the `PartialOrd` trait, especially with the word "implements".

Maybe something like "compares the `PartialOrd` elements of this"?

Also, while you're here, it would be great to have some prose about what happens for incomparable items like NANs.

Reply:
@scottmcm, thanks.  I agree that adding an explanation about `PartialOrd` elements is correct.  I also added some prose about using `partial_cmp` with things like `NANs`.